File: src/main/java/com/taykt/api/examples/webhook/EchoServlet.java
package com.taykt.api.examples.webhook;

import java.io.IOException;

import javax.servlet.http.HttpServlet;
import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;

public class EchoServlet extends HttpServlet {
	
	private static final long serialVersionUID = -8879998764264360723L;

	public void doPost(HttpServletRequest req, HttpServletResponse resp) throws IOException {
    	
    	String from = req.getParameter("pid");
    	String mo_msg = req.getParameter("text");
    	
    	resp.setContentType("text/plain");
    	resp.setCharacterEncoding("utf-8");	
    	resp.getWriter().println( from + " sent " + mo_msg  );
    }
   
}

File: src/main/scala/com/taykt/api/examples/webhook/EchoServlet.scala
package com.taykt.api.examples.webhook

import javax.servlet.http.{HttpServlet, HttpServletRequest, HttpServletResponse}

/**
 * A Taykt API web hook which accepts a text message and sends back an SMS
 * message that echos what was sent. More-or-less.
 * 
 * See: http://api.taykt.com/ for details.
 */
class EchoServlet extends HttpServlet {
	
	override def doPost(req:HttpServletRequest, resp:HttpServletResponse) {
   
		// Extract details sent from Taykt to this web hook:
		val from = req.getParameter("pid")
    	val mo_msg = req.getParameter("text")
  
    	// Compose and send the response:
    	resp.setContentType("text/plain")
    	resp.setCharacterEncoding("utf-8");
    	resp.getWriter().println( from + " sent " + mo_msg  )
    }
  
}
